plasma_framework/contracts/src/vaults/Erc20Vault.sol

```diff
@@ -49,7 +50,7 @@ contract Erc20Vault is Vault {
     * @param token address of ERC20 token contract.
     * @param amount amount to transfer.
     */
-    function withdraw(address payable receiver, address token, uint256 amount) external onlyFromNonQuarantinedExitGame {
+    function withdraw(address payable receiver, address token, uint256 amount) external onlyFromNonQuarantinedExitGame gracefullyNonReentrant {
```

Comment:
This could possibly be just `nonReentrant` as if token transfer fails then most probably token contract is broken and we won't be able to process exitQueue for that token anyway.